src/python/gem5/simulate/exit_handler.py

```diff
@@ -0,0 +1,207 @@
+                simulator._on_exit_event[exit_enum]
+            )
+        except StopIteration:
+            # If the user's generator has ended, throw a warning and use
+            # the default generator for this exit event.
+            warn(
+                "User-specified generator/function list for the exit "
+                f"event'{exit_enum.value}' has ended. Using the default "
+                "generator."
+            )
+            self._exit_on_completion = next(
+                simulator._default_on_exit_dict[exit_enum]
+            )
+
+        except KeyError:
+            # If the user has not specified their own generator for this
+            # exit event, use the default.
+            self._exit_on_completion = next(
+                simulator._default_on_exit_dict[exit_enum]
+            )
+
+    @overrides(ExitHandler)
+    def _exit_simulation(self) -> bool:
+        assert (
+            self._exit_on_completion is not None
+        ), "Exit on completion boolean var is not set."
+        return self._exit_on_completion
+
+
+class ScheduledExitEventHandler(ExitHandler):
+    """A handler designed to be the default for  an Exit scheduled to occur
+    at a specified tick. For example, these Exit exits can be triggered through be
+    src/python/m5/simulate.py's `scheduleTickExitFromCurrent` and
+    `scheduleTickExitAbsolute` functions.
+
+    It will exit the simulation loop by default.
+
+    The `justification` and `scheduled_at_tick` methods are provided to give
+    richer information about this schedule exit.
+    """
+
+    @overrides(ExitHandler)
+    def _process(self, simulator: "Simulator") -> None:
+        pass
+
+    def justification(self) -> Optional[str]:
+        """Returns the justification for the scheduled exit event.
+
+        This information may be passed to when scheduling the event event to
+        remind the user why the event was scheduled, or used in cases where
+        lots of scheduled events are used and there is a need to differentiate
+        them.
+
+        Returns "None" if this information was not set.
+        """
+        return (
+            None
+            if "justification" not in self._payload
+            else self._payload["justification"]
+        )
+
+    def scheduled_at_tick(self) -> Optional[int]:
+        """Returns the tick in which the event was scheduled on (not scheduled
+        to occur, but the tick the event was created).
+
+        Returns "None" if this information is not available.
+        """
+        return (
+            None
+            if "scheduled_at_tick" not in self._payload
+            else int(self._payload["scheduled_at_tick"])
+        )
+
+    @overrides(ExitHandler)
+    def _exit_simulation(self) -> bool:
+        return True
+
+
+class KernelBootedExitHandler(ExitHandler):
+    @overrides(ExitHandler)
+    def _process(self, simulator: "Simulator") -> None:
+        print("First exit: kernel booted")
```

Comment:
"First" and "Second" (below) should probably be removed. We can't guarantee that there won't be some new event added before this.

Also, we should probably not use "print" but some kind of logging. Maybe this is a change for another patch, though.